src/__swaps__/screens/Swap/components/CoinRow.tsx

```diff
@@ -11,7 +11,104 @@ import Animated from 'react-native-reanimated';
 import { StyleSheet } from 'react-native';
 import { useSwapContext } from '@/__swaps__/screens/Swap/providers/swap-provider';
 import { SwapCoinIcon } from './SwapCoinIcon';
-import { ethereumUtils } from '@/utils';
+import { ethereumUtils, haptics, showActionSheetWithOptions } from '@/utils';
+import { ContextMenuButton } from 'react-native-ios-context-menu';
+import { IS_ANDROID } from '@/env';
+import { startCase } from 'lodash';
+import { setClipboard } from '@/hooks/useClipboard';
+import { RainbowNetworks } from '@/networks';
+import * as i18n from '@/languages';
+
+const InfoButton = ({ address, chainId }: { address: string; chainId: ChainId }) => {
+  const network = RainbowNetworks.find(network => network.id === chainId)?.value;
+
+  const handleCopy = useCallback(() => {
+    haptics.selection();
+    setClipboard(address);
+  }, [address]);
+
+  const options = {
+    copy: {
+      title: i18n.t(i18n.l.exchange.coin_row.copy_contract_address),
+      action: handleCopy,
+    },
+    ...(network
+      ? {
+          blockExplorer: {
+            title: i18n.t(i18n.l.exchange.coin_row.view_on, { blockExplorerName: startCase(ethereumUtils.getBlockExplorer(network)) }),
+            action: () => ethereumUtils.openAddressInBlockExplorer(address, network),
+          },
+        }
+      : {}),
+  };
+
+  const menuConfig = {
+    menuItems: [
+      {
+        actionKey: 'copyAddress',
+        actionTitle: options.copy.title,
+        icon: {
+          iconType: 'SYSTEM',
+          iconValue: 'doc.on.doc',
+        },
+      },
+      ...(network
+        ? [
+            {
+              actionKey: 'blockExplorer',
+              actionTitle: options.blockExplorer?.title,
+              icon: {
+                iconType: 'SYSTEM',
+                iconValue: 'link',
+              },
+            },
+          ]
+        : []),
+    ],
+    menuTitle: '',
+  };
+
+  const handlePressMenuItem =
+    // @ts-expect-error ContextMenu is an untyped JS component and can't type its onPress handler properly
```

Comment:
I see this multiple areas in our app and more of a nitpick, but the proper type is `OnPressMenuItemEventObject`.